### PR TITLE
Fix tests after validator logic change

### PR DESCRIPTION
**Pull request by emailtovamos**
This includes addressing of most of the suggestions from @Ullaakut from https://github.com/flare-foundation/flare/pull/25 but it is missing rebasing with master. 
I created this new PR as the older PR's rebasing didn't work properly.

### CODE REVIEW

File: network/network_test.go [hunk 1]
```diff
@@ -1467,11 +1465,11 @@ func TestPeerAliasesDisconnect(t *testing.T) {
 	)
 	id2 := ids.ShortID(hashing.ComputeHash160Array([]byte(ip2.IP().String())))
 
-	vdrs := validators.NewManager(0,
-		validators.WithValidator(id0, 1),
-		validators.WithValidator(id1, 1),
-		validators.WithValidator(id2, 1),
-	)
+	//vdrs := validators.NewDefaultSet(constants.CostonID)
```
Comment (Ullaakut) [inline network/network_test.go:1468]: Why was this kept commented out? Should be removed unless there's a good reason for it IMO.

File: network/network_test.go [hunk 2]
```diff
@@ -9,12 +9,13 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"errors"
-	"math"
 	"net"
 	"sync"
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common/math"
```
Comment (Ullaakut) [inline network/network_test.go:17]: If this one was imported just for `MaxUint64`, we should use the native `math` library instead.

File: snow/validators/test_retriever.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	"github.com/flare-foundation/flare/ids"
+)
+
+type RetrieverMock struct {
+	GetValidatorsByBlockIDFunc func(blockID ids.ID) (Set, error)
+}
+
+func (m *RetrieverMock) GetValidatorsByBlockID(blockID ids.ID) (Set, error) {
+	return m.GetValidatorsByBlockIDFunc(blockID)
+
```
Comment (Ullaakut) [inline snow/validators/test_retriever.go:13]:
```suggestion
```

File: snow/validators/test_updater.go [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package validators
+
+import (
+	"github.com/flare-foundation/flare/ids"
+)
+
+type UpdaterMock struct {
+	UpdateValidatorsF func(blockID ids.ID) error
```
Comment (Ullaakut) [inline snow/validators/test_updater.go:8]:
```suggestion
	UpdateValidatorsFunc func(blockID ids.ID) error
```

File: snow/validators/test_updater.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+func (m *UpdaterMock) UpdateValidators(blockID ids.ID) error {
+	return m.UpdateValidatorsF(blockID)
+
+}
```
Comment (Ullaakut) [inline snow/validators/test_updater.go:14]:
```suggestion
func (m *UpdaterMock) UpdateValidators(blockID ids.ID) error {
	return m.UpdateValidatorsFunc(blockID)
}
```

File: snow/validators/test_vmstate.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	GetCurrentHeightF func() (uint64, error)
+}
+
+func (T *TestVMState) GetCurrentHeight() (uint64, error) {
+	if T.GetCurrentHeightF != nil {
+		return T.GetCurrentHeightF()
+	}
+
+	return 0, errGetCurrentHeight
+}
```
Comment (Ullaakut) [inline snow/validators/test_vmstate.go:24]:
```suggestion
type VMStateMock struct {
	GetCurrentHeightF func() (uint64, error)
}

func (v *VMStateMock) GetCurrentHeight() (uint64, error) {
	return v.GetCurrentHeightFunc()
}
```

File: snow/validators/test_vmstate.go [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package validators
+
+import (
+	"errors"
+	"testing"
+)
+
+var (
+	errGetCurrentHeight = errors.New("unexpectedly called GetCurrentHeight")
+)
```
Comment (Ullaakut) [inline snow/validators/test_vmstate.go:10]:
```suggestion
```

File: snow/validators/test_retriever.go [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package validators
```
Comment (Ullaakut) [inline snow/validators/test_retriever.go:1]: Also I just realized that those mocks are exposed in the package. They should not. This means that anyone using the `validators` package will see the mocks in auto-complete, and be able to use them in normal code.

They should either be defined in a `mocks_test.go` file, or even better would be in a package called `validators_test` within the same directory.

File: vms/proposervm/batched_vm_test.go [hunk 9]
```diff
@@ -890,12 +890,32 @@ func initTestRemoteProposerVM(
...
+			return nil
+		},
+	}
+	retriever := &validators.RetrieverMock{
+		GetValidatorsByBlockIDFunc: func(blockID ids.ID) (validators.Set, error) {
+			s := validators.NewSet()
+			s.AddWeight(ids.ShortID{11}, 3)
+			return s, nil
+		},
+	}
```
Comment (Ullaakut) [inline vms/proposervm/batched_vm_test.go:910]: Should be updated following my previous suggestions

File: vms/proposervm/block.go [hunk 10]
```diff
@@ -82,9 +82,6 @@ func (p *postForkCommonComponents) Verify(parentTimestamp time.Time, parentPChai
 	}
 
 	childPChainHeight := child.PChainHeight()
-	if childPChainHeight < parentPChainHeight {
-		return errPChainHeightNotMonotonic
-	}
```
Comment (Ullaakut) [inline vms/proposervm/block.go:87]: Why is this removed?

File: vms/proposervm/post_fork_block_test.go [hunk 11]
```diff
@@ -890,8 +890,8 @@ func TestBlockAccept_PostForkBlock_TwoProBlocksWithSameCoreBlock_OneIsAccepted(t
 	if err != nil {
 		t.Fatal("could not build proBlk2")
 	}
-	if proBlk1.ID() == proBlk2.ID() {
-		t.Fatal("proBlk1 and proBlk2 should be different for this test")
+	if proBlk1.ID() != proBlk2.ID() {
+		t.Fatal("proBlk1 and proBlk2 should not be different for this test")
```
Comment (Ullaakut) [inline vms/proposervm/post_fork_block_test.go:894]: I don't recall if you answered about this in the previous PR, so my question still stands: Why was this test inverted? It's usually not a great sign when the only change to the test is that we now expect the opposite to happen 😄 It feels to me like this means that the original stuff we were testing is no longer true, so there was a breaking change, and we should write another test to replace it, to make sure that the new test covers the new behavior.

**Review by Ullaakut — CHANGES_REQUESTED** [review]

File: vms/proposervm/block.go (hunk 10 above)
Comment (emailtovamos) [reply vms/proposervm/block.go:87]: The p-chain height doesn't change anymore. So this check is no longer required.

File: vms/proposervm/post_fork_block_test.go (hunk 11 above)
Comment (emailtovamos) [reply vms/proposervm/post_fork_block_test.go:894]: I am not fully sure what is the best thing to do here. `pChainHeight` was increased and a block was created. But since it is no longer supposed to increase, it is not surprising why `proBlk1.ID() == proBlk2.ID()` holds true.  
I should ask @awfm9

File: network/network_test.go [hunk 12]
```diff
@@ -2905,10 +2896,7 @@ func TestAppGossip(t *testing.T) {
 	caller0.outbounds[ip2.IP().String()] = listener2
 	caller1.outbounds[ip2.IP().String()] = listener2
 
-	vdrs := validators.NewManager(
-		0,
-		validators.WithValidator(id2, math.MaxUint64), // id2 is a validator
-	)
+	vdrs := validators.NewSet()
```
Comment (awfm9) [inline network/network_test.go:2899]: This is missing `id2` being added.

File: network/network_test.go [hunk 13]
```diff
@@ -2667,10 +2661,7 @@ func TestPeerGossip(t *testing.T) {
...
```
Comment (awfm9) [inline network/network_test.go:2664]: This is missing `id2` being added.

File: vms/proposervm/block.go (hunk 10 above)
Comment (awfm9) [reply vms/proposervm/block.go:87]: It should still not be removed. We talked about this. We are not touching P-Chain related logic for now.

File: vms/proposervm/post_fork_block_test.go (hunk 11 above)
Comment (awfm9) [reply vms/proposervm/post_fork_block_test.go:894]: This is definitely a mistake. This is like fixing a test that suddenly breaks by expecting it to error. Even though it shouldn't.

**Review by awfm9 — CHANGES_REQUESTED** [review]
There are still issues with this PR.

File: snow/validators/test_retriever.go (hunk 8 above)
Comment (awfm9) [reply snow/validators/test_retriever.go:4]: This is based on how it was done by Avalanche before, so I agree with his choice to keep it like this.
